app/index.js

```diff
@@ -64,6 +65,14 @@ const contentSettings = require('../js/state/contentSettings')
 const privacy = require('../js/state/privacy')
 const basicAuth = require('./browser/basicAuth')
 
+if (!process.env.USER_DATA_DIR && ['development', 'test'].includes(process.env.NODE_ENV)) {
+  process.env.USER_DATA_DIR = path.join(app.getPath('appData'), app.getName() + '-' + process.env.NODE_ENV)
+}
+
+if (process.env.USER_DATA_DIR) {
```

Comment:
Actually sorry, I think we should prefix this env var with BRAVE. There's a chance this will be defined on some people's machines already and cause problems.